lib/appmap/agent.rb

```diff
@@ -0,0 +1,115 @@
+require_relative 'version'
+require_relative 'hook'
+require_relative 'config'
+require_relative 'trace'
+require_relative 'class_map'
+require_relative 'metadata'
+require_relative 'util'
+require_relative 'open'
+
+# load extension
+require_relative 'appmap'
+
+module AppMap
```

Comment:
Was this just extracted verbatim from lib/appmap.rb? Are there any changes in here that need to be reviewed>

Reply:
The only changes was to use `Util.deep_dup` instead of ActiveSupport core_ext `deep_dup`. In Util, it does a 2-way Marshal instead of traversing through all the data. Since that works fine for this case and performance doesn't much matter.